axlearn/open_api/openai.py

```diff
@@ -77,11 +78,13 @@ async def async_generate(
                     **kwargs,
                 )
             else:
+                req_kwargs = copy.deepcopy(kwargs)
```

Comment:
we do a full copy here since kwargs would be reused for retry. Hence we do not want to modify it here just due to `tools` field.